From 0d685f4ec8b54860a031fe3e04a2fff535288f80 Mon Sep 17 00:00:00 2001
From: Ankur Tyagi <ankur.tyagi85@gmail.com>
Date: Mon, 26 Jan 2026 22:46:51 +1300
Subject: [PATCH] raid6check.c, restripe.c: Use 64-bit off_t across both musl
 and glibc

This commit is adaptation of original patch[1] after commit[2] addressed
lseek issues.

[1] https://lore.kernel.org/linux-raid/20221110225546.337164-1-raj.khem@gmail.com/
[2] https://github.com/md-raid-utilities/mdadm/commit/787cc1b60130b8031be59e49d54463c58cd8cf74

Upstream-Status: Submitted [https://github.com/md-raid-utilities/mdadm/pull/235]
Signed-off-by: Ankur Tyagi <ankur.tyagi85@gmail.com>
---
 raid6check.c | 4 ++--
 restripe.c   | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/raid6check.c b/raid6check.c
index 4469dc8f..fa50d44e 100644
--- a/raid6check.c
+++ b/raid6check.c
@@ -288,7 +288,7 @@ int manual_repair(int chunk_size, int syndrome_disks,
 	}
 
 	int write_res1, write_res2;
-	off64_t seek_res;
+	off_t seek_res;
 
 	seek_res = lseek(source[fd1], offsets[fd1] + start * chunk_size, SEEK_SET);
 	if (seek_res < 0) {
@@ -381,7 +381,7 @@ int check_stripes(struct mdinfo *info, int *source, unsigned long long *offsets,
 			goto exitCheck;
 		}
 		for (i = 0 ; i < raid_disks ; i++) {
-			off64_t seek_res = lseek(source[i], offsets[i] + start * chunk_size,
+			off_t seek_res = lseek(source[i], offsets[i] + start * chunk_size,
 						   SEEK_SET);
 			if (seek_res < 0) {
 				fprintf(stderr, "lseek to source %d failed\n", i);
diff --git a/restripe.c b/restripe.c
index ec8d6275..58c0f5f5 100644
--- a/restripe.c
+++ b/restripe.c
@@ -756,7 +756,7 @@ int restore_stripes(int *dest, unsigned long long *offsets,
 			if (src_buf == NULL) {
 				/* read from file */
 				if (lseek(source, read_offset, 0) !=
-					 (off64_t)read_offset) {
+					 (off_t)read_offset) {
 					rv = -1;
 					goto abort;
 				}
